miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/ExperimentListConfiguration.java

```diff
@@ -0,0 +1,143 @@
+package uk.ac.bbsrc.tgac.miso.webapp.util;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Library;
+import uk.ac.bbsrc.tgac.miso.core.data.Pair;
+import uk.ac.bbsrc.tgac.miso.core.data.Partition;
+import uk.ac.bbsrc.tgac.miso.core.data.Platform;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.PoolableElementView;
+import uk.ac.bbsrc.tgac.miso.core.util.WhineyFunction;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.ExperimentDto;
+import uk.ac.bbsrc.tgac.miso.dto.ExperimentDto.RunPartitionDto;
+import uk.ac.bbsrc.tgac.miso.dto.PartitionDto;
+import uk.ac.bbsrc.tgac.miso.dto.PlatformDto;
+import uk.ac.bbsrc.tgac.miso.dto.RunDto;
+import uk.ac.bbsrc.tgac.miso.dto.StudyDto;
+import uk.ac.bbsrc.tgac.miso.service.ExperimentService;
+import uk.ac.bbsrc.tgac.miso.service.LibraryService;
+
+public class ExperimentListConfiguration implements Serializable {
+  public static class AddRequest {
+    private ExperimentDto experiment;
+    private PartitionDto partition;
+
+    public ExperimentDto getExperiment() {
+      return experiment;
+    }
+
+    public PartitionDto getPartition() {
+      return partition;
+    }
+
+    public void setExperiment(ExperimentDto experiment) {
+      this.experiment = experiment;
+    }
+
+    public void setPartition(PartitionDto partition) {
+      this.partition = partition;
+    }
+  }
+
+  public static class StudiesForExperiment {
+    private ExperimentDto experiment;
+    private List<StudyDto> studies;
+
+    public ExperimentDto getExperiment() {
+      return experiment;
+    }
+
+    public List<StudyDto> getStudies() {
+      return studies;
+    }
+
+    public void setExperiment(ExperimentDto experiment) {
+      this.experiment = experiment;
+    }
+
+    public void setStudies(List<StudyDto> studies) {
+      this.studies = studies;
+    }
+  }
+
+  private static final long serialVersionUID = 1L;
+
+  private final List<AddRequest> addToExperiment;
+  private final List<StudiesForExperiment> studiesForExperiment;
+  private final long runId;
+
+  public ExperimentListConfiguration(ExperimentService experimentService, LibraryService libraryService, Platform platform, Run run) {
+    runId = run.getId();
+    PlatformDto platformDto = Dtos.asDto(platform);
+    RunDto runDto = Dtos.asDto(run);
+
+    // Gather all the libraries in this run and then build a map from library to all the partitions that hold it in this run
+    Map<Library, List<Partition>> libraryGroups = run.getSequencerPartitionContainers().stream()//
+        .flatMap(container -> container.getPartitions().stream())//
+        .filter(partition -> partition.getPool() != null)//
+        .flatMap(partition -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)